Lesson31/Kubra/my-app/src/components/List/index.jsx

```diff
@@ -1,34 +1,56 @@
-import {ListItem} from '../ListItem';
-import {useState} from "react";
-import './styles.css';
+import { ListItem } from "../ListItem";
+import { useState } from "react";
+import "./styles.css";
+import { NewListItem } from "../NewListItem";
 
 export const List = () => {
-  const [data, setData]  = useState([
-    {id: 1, title: "Do dishes", completed: false},
-    {id: 2, title: "Do homework", completed: true},
-    {id: 3, title: "Go running", completed: false},
-    {id: 4, title: "Do dishes 2", completed: false},
-    {id: 5, title: "Do homework 3", completed: false},
-    {id: 6, title: "Go running 4", completed: false},
+  const [data, setData] = useState([
+    { id: 1, title: "Do dishes", completed: false },
+    { id: 2, title: "Do homework", completed: true },
+    { id: 3, title: "Go running", completed: false },
+    { id: 4, title: "Do dishes 2", completed: false },
+    { id: 5, title: "Do homework 3", completed: false },
+    { id: 6, title: "Go running 4", completed: false },
   ]);
 
   const handleClick = (id) => {
     setData((prevData) => {
-      return prevData.map(dataItem => 
-        dataItem.id === id ? {...dataItem, completed: !dataItem.completed} : dataItem)
+      return prevData.map((dataItem) =>
+        dataItem.id === id
+          ? { ...dataItem, completed: !dataItem.completed }
+          : dataItem
+      );
     });
   };
-  
-return (
-  <ul className="todo-list">
-      {
-        data.map(listItem => 
-        <ListItem key={listItem.id}
-        id={listItem.id}
-        title={listItem.title}
-        completed={listItem.completed}
-        handleClick={handleClick}/>)
-      }
-  </ul>
-)
-};
\ No newline at end of file
+
+  const addNewItem = (text) => {
+    setData([...data, { id: Date.now(), title: text, completed: false }]);
```

Comment:
I liked the solution that you used for creating a new id very much. It's the best approach I saw so far in both this group and the second group. Using `Date.now()` would make sure that a unique long id is created and since miliseconds are taken into account it would be almost impossible to duplicate the id. Well done!